### ISSUE TITLE
Improve functionality of ElectrodeCategoryReader

### ISSUE THREAD

**Issue by leondavis1** [pennmem/cmlreaders#72]
The electrode category file is highly irregular, but we do need to be able to consistently read it in a reasonable form

**Comment by mivade** [comment]
What's wrong with the current reader for it? It handles a lot of the irregularities.

https://github.com/pennmem/cmlreaders/blob/2e96edad003ad159ac5b6375575ce7fe109970b5/cmlreaders/readers/readers.py#L246-L252

**Comment by esolomon** [comment]
@mivade Having looked at my personal code that handles electrode categories, I think providing you with some pseudocode would honestly be more useful: 

###
For each line of electrode_categories.txt to be "decoded": 
if 'seizure' in txt.lower() or 'soz' in txt.lower():
Classify as "Seizure Onset Zone"

if 'ictal' in txt.lower():
Classify as 'Interictal Spiking'

if 'bad' in txt.lower() or 'broken' in txt.lower():
Classify as "Bad Electrode"
###

In other words, search for ('seizure', 'soz') to find Seizure Onset Zone labels, search for ('ictal') to find Interictal Spiking electrodes, and search for ('broken', 'bad') to find Bad Electrodes.
